DFC.Digital/DFC.Digital.Web.Sitefinity.CourseModule/Mvc/Views/CourseDetails/CourseDetails.cshtml

```diff
@@ -1,126 +1,194 @@
                     <td class="govuk-table__cell">@Model.QualificationLevel</td>
                 </tr>
+            }
+            @if (!string.IsNullOrWhiteSpace(Model.EntryRequirements))
+            {
                 <tr class="govuk-table__row">
                     <th class="govuk-table__header" scope="row">Entry requirements</th>
                     <td class="govuk-table__cell">@Model.EntryRequirements</td>
                 </tr>
+            }
+            @if (!string.IsNullOrWhiteSpace(Model.StartDateLabel))
+            {
                 <tr class="govuk-table__row">
                     <th class="govuk-table__header" scope="row">Start date</th>
                     <td class="govuk-table__cell">@Model.StartDateLabel</td>
                 </tr>
+            }
+            @if (!string.IsNullOrWhiteSpace(Model.Cost))
+            {
                 <tr class="govuk-table__row">
                     <th class="govuk-table__header" scope="row">Cost</th>
                     <td class="govuk-table__cell">@Model.Cost</td>
                 </tr>
+            }
+            @if (!string.IsNullOrWhiteSpace(Model.Duration))
+            {
                 <tr class="govuk-table__row">
                     <th class="govuk-table__header" scope="row">Duration</th>
                     <td class="govuk-table__cell">@Model.Duration</td>
                 </tr>
+            }
+            @if (!string.IsNullOrWhiteSpace(Model.AttendanceMode))
+            {
                 <tr class="govuk-table__row">
                     <th class="govuk-table__header" scope="row">Classroom or online</th>
                     <td class="govuk-table__cell">@Model.AttendanceMode</td>
                 </tr>
+            }
+            @if (!string.IsNullOrWhiteSpace(Model.AttendancePattern))
+            {
                 <tr class="govuk-table__row">
                     <th class="govuk-table__header" scope="row">Day, night or weekend</th>
                     <td class="govuk-table__cell">@Model.AttendancePattern</td>
                 </tr>
-            </tbody>
-        </table>
+            }
+        </tbody>
+    </table>
 
-        <a href="#" role="button" draggable="false" class="govuk-button govuk-button--start govuk-!-margin-bottom-2">
-            Enrol now
-        </a>
-        <p class="govuk-body govuk-!-font-size-14 govuk-!-margin-0">on the provider's website</p>
+    <a href="#" role="button" draggable="false" class="govuk-button govuk-button--start govuk-!-margin-bottom-2">
+        Enrol now
+    </a>
 
-        <h2 class="govuk-heading-m">Course description</h2>
-        <p class="govuk-body">@Model.Description</p>
+    <p class="govuk-body govuk-!-font-size-14 govuk-!-margin-0">on the provider's website</p>
 
-        <h2 class="govuk-heading-m">Entry requirements</h2>
-        <p class="govuk-body">@Model.EntryRequirements</p>
+    <h2 class="govuk-heading-m">Course description</h2>
```

Comment:
All these headings should be properties of the widget

Reply:
done